Content.Server/Health/BodySystem/Mechanism/DroppedMechanismComponent.cs

```diff
@@ -8,32 +8,184 @@
 using Robust.Shared.ViewVariables;
 using System.Globalization;
 using Robust.Server.GameObjects;
+using Content.Server.GameObjects.EntitySystems;
+using Robust.Shared.Log;
+using Content.Shared.Interfaces;
+using Robust.Server.GameObjects.Components.UserInterface;
+using Robust.Server.Interfaces.Player;
+using Robust.Shared.Interfaces.Random;
+using Robust.Shared.Interfaces.GameObjects;
+using System.Diagnostics;
 
 namespace Content.Server.BodySystem {
 
     /// <summary>
     ///    Component containing the data for a dropped Mechanism entity.
     /// </summary>
     [RegisterComponent]
-    public class DroppedMechanismComponent : Component
+    public class DroppedMechanismComponent : Component, IAfterInteract
     {
+
+#pragma warning disable 649
+        [Dependency] private readonly ISharedNotifyManager _sharedNotifyManager;
+        [Dependency] private readonly IRobustRandom _random;
+        [Dependency] private IPrototypeManager _prototypeManager;
+#pragma warning restore 649
+
         public sealed override string Name => "DroppedMechanism";
 
         [ViewVariables]
-        private Mechanism _containedMechanism;
+        public Mechanism ContainedMechanism { get; private set; }
 
-        public Mechanism ContainedMechanism => _containedMechanism;
+        private BoundUserInterface _userInterface;
+        private Dictionary<object, object> _optionsCache = new Dictionary<object, object>();
```

Comment:
Any reason this isn't an `<int, BodyPart>` dictionary? It looks from the code that at least `<int,object>` should be possible.

Reply:
I had it using both integer and string keys in the past, but I've changed it now to be an int dictionary.